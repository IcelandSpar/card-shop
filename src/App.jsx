import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      <h1>{count}</h1>
      <button onClick={() => {
        setCount(prev => prev + 1)
      }}>Increment</button>
      <p>Hello World</p>
      
    </>
  )
}

export default App
