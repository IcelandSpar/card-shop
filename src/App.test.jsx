import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import App from './App.jsx';


it('expect true to be true', () => {
    expect(true).toBe(true);
})

it('expect to render', () => {
    render(<App/>);
    const greetingElement = screen.getByText('Hello World');
    expect(greetingElement).toBeInTheDocument();
    
})

it('count is 0', () => {
    render(<App/>);
    const headerCount = screen.getByRole('heading', {name: '0'})
    expect(headerCount).toHaveTextContent('0');
})

it('count is 1 on button click', async () => {
    const user = userEvent.setup();
    render(<App/>);
    const headerCount = screen.getByRole('heading')
    const buttonIncrement = screen.getByRole('button', {name: 'Increment'});
    await user.click(buttonIncrement);
    await user.click(buttonIncrement);
    expect(headerCount).toHaveTextContent('2');



})

