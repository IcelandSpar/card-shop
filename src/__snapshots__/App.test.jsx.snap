// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect to render 1`] = `
<p>
  Hello World
</p>
`;
